src/contracts/pods/EigenPodManager.sol

```diff
@@ -134,68 +184,142 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
-     * @param amount The amount of ETH to 'deposit' (i.e. be credited to the podOwner).
+     * @param amountWei The amount of ETH to 'deposit' (i.e. be credited to the podOwner).
      * @dev Callable only by the podOwner's EigenPod contract.
      */
-    function restakeBeaconChainETH(address podOwner, uint256 amount) external onlyEigenPod(podOwner) {
-        strategyManager.depositBeaconChainETH(podOwner, amount);
-        emit BeaconChainETHDeposited(podOwner, amount);
+    function restakeBeaconChainETH(address podOwner, uint256 amountWei) 
+        external 
+        onlyEigenPod(podOwner) 
+        onlyWhenNotPaused(PAUSED_DEPOSITS)
+        onlyNotFrozen(podOwner)
+        nonReentrant
+    {
+        require(podOwner != address(0), "EigenPodManager.restakeBeaconChainETH: podOwner cannot be zero address");
+        require(amountWei > 0, "EigenPodManager.restakeBeaconChainETH: amount must be greater than zero");
+
+        podOwnerShares[podOwner] += amountWei;     
+
+        //if the podOwner has delegated shares, record an increase in their delegated shares
+        delegationManager.increaseDelegatedShares(podOwner, beaconChainETHStrategy, amountWei);
+        emit BeaconChainETHDeposited(podOwner, amountWei);
     }
 
     /**
      * @notice Removes beacon chain ETH from EigenLayer on behalf of the owner of an EigenPod, when the
      *         balance of a validator is lower than how much stake they have committed to EigenLayer
      * @param podOwner is the pod owner whose balance is being updated.
-     * @param beaconChainETHStrategyIndex is the index of the beaconChainETHStrategy in case it must be removed,
      * @param sharesDelta is the change in podOwner's beaconChainETHStrategy shares
      * @dev Callable only by the podOwner's EigenPod contract.
      */
-    function recordBeaconChainETHBalanceUpdate(address podOwner, uint256 beaconChainETHStrategyIndex, int256 sharesDelta) external onlyEigenPod(podOwner) {
-        strategyManager.recordBeaconChainETHBalanceUpdate(podOwner, beaconChainETHStrategyIndex, sharesDelta);
+    function recordBeaconChainETHBalanceUpdate(address podOwner, int256 sharesDelta) external onlyEigenPod(podOwner) nonReentrant {
+       _updateSharesToReflectBeaconChainETHBalance(podOwner, sharesDelta);
     }
 
     /**
-     * @notice Withdraws ETH from an EigenPod. The ETH must have first been withdrawn from the beacon chain.
-     * @param podOwner The owner of the pod whose balance must be withdrawn.
-     * @param recipient The recipient of the withdrawn ETH.
-     * @param amount The amount of ETH to withdraw.
-     * @dev Callable only by the StrategyManager contract.
+     * @notice Queues a withdrawal for beacon chain ETH shares on behalf of the owner of an EigenPod.
+     * @param amountWei The amount of ETH to withdraw.
+     * @param undelegateIfPossible If true, the podOwner's shares will be undelegated if they are not currently delegated.
```

Comment:
I'm worried by this description. it is either inaccurate or there is a bug in the system.

Reply:
resolved this with changes myself